src/lockfile/index.js

```diff
@@ -102,6 +124,20 @@ export default class Lockfile {
 
   parseResultType: ?ParseResultType;
 
+  // if true, we're parsing an old yarn file and need to update integrity fields
+  entriesExistWithoutIntegrity(): boolean {
+    if (!this.cache) {
+      return false;
+    }
+    return Object.keys(this.cache).some(key => {
+      if (this.cache && this.cache[key] && !/^.*@(file:|http)/.test(key)) {
+        return !this.cache[key].integrity;
+      } else {
+        return false; // some entries are not supposed to have integrity, eg. files and web links
```

Comment:
Why's that?

Reply:
I'll start with the short and honest answer: because it would require a great deal more changes than this already monstrous and hard-to-swallow PR. :)
Right now, what this PR does is create compatibility with the npm integrity feature. It mirrors the integrity from the npm repository (and creates it when publishing to npm through yarn). All of the integrity checks are performed against fields that at some point were in a (npm) registry json file. This whole feature is built to support that. For everything else, we still have the `hash` field which - while not perfect - can keep serving more than just backwards compatibility for now. This feature does give a good infrastructure to eventually replace the `hash` field (maybe in a new lockfile version?) though. When it does that, it can replace all the other roles of the `hash` field such as being a checksum for non-registry packages.
I hope we see eye to eye on this one?

Reply:
That sounds fair.